src/components/ha-form/ha-form-string.ts

```diff
@@ -33,7 +35,7 @@ export class HaFormString extends LitElement implements HaFormElement {
   }
 
   protected render(): TemplateResult {
-    return this.schema.name.includes("password")
+    return SECRET_FIELDS.includes(this.schema.name)
```

Comment:
Old check was "name includes password", new check is "name == password". We should do the original check again.